### PR TITLE
added notes for how to use minio as eucalyptus object storage

### PR DESCRIPTION
**Pull request by shaon**

### CODE REVIEW

**Comment by harshavardhana** [comment]
Hi @shaon - sorry for our late reply can you fix these warnings in your doc? 

```sh
$ remark -q . 
docs/minio-as-eucalyptus-object-storage.md
       3:1-3:30  warning  Use the trailing [] on reference links         no-shortcut-reference-link  remark-lint
       3:1-3:30  warning  Found reference to undefined definition        no-undefined-references     remark-lint
       5:1-5:39  warning  Use the trailing [] on reference links         no-shortcut-reference-link  remark-lint
       5:1-5:39  warning  Found reference to undefined definition        no-undefined-references     remark-lint
  115:45-115:87  warning  Don’t use literal URLs without angle brackets  no-literal-urls             remark-lint
  116:48-116:98  warning  Don’t use literal URLs without angle brackets  no-literal-urls             remark-lint
   120:1-120:45  warning  Don’t use literal URLs without angle brackets  no-literal-urls             remark-lint
```

File: docs/minio-as-eucalyptus-object-storage.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# euctl -r objectstorage.providerclient
+euctl: error (ModifyPropertyValueType): Cannot modify \
+objectstorage.providerclient.providerclient new value \
+is not a valid value.  Legal values are: walrus,ceph-rgw,riakcs
+```
+
+So, now we have to add another provider client for Minio. Technically, we can probably use riakcs, if you have deployed Eucalyptus with packages, but I am not gonna do that in this case, since I already have source build cloud.
+
+This post assumes that you know how to build Eucalyptus from source and doesn't go into much detail. The purpose of this post is to show how to add a third-party object storage like Minio with Eucalyptus. But again, feel free to use package installation and use riakcs as provider client, use the Minio's endpoint and user credentials as described below.
+
```
Comment (NitishT) [inline docs/minio-as-eucalyptus-object-storage.md]: Please add a link that details the Eucalyptus build process.

File: docs/minio-as-eucalyptus-object-storage.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+@ObjectStorageProviderClientProperty("minio")
+public class MinioProviderClient extends S3ProviderClient {
+}
+```
+
+Build and install all of eucalyptus source or just build the jars.
+
+Stop eucalyptus-cloud service and copy eucalyptus-object-storage-4.4.0.jar to /usr/share/eucalyptus directory where user-facing services are running. Restart eucalyptus-cloud service.
+
+If everything goes well, when the services are enabled, we can now check if the supported object storage providers again.
```
Comment (NitishT) [inline docs/minio-as-eucalyptus-object-storage.md]: Sentence `we can now check if the supported object storage providers again` is not clear. Please reword.

File: docs/minio-as-eucalyptus-object-storage.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+Now we can use Minio as an object storage backend. However, while working, I found an issue where any PUT object request that was made through Eucalyptus ends up with a failure. After debugging with Swathi Gangisetty it looks like the AWS Java SDK Eucalyptus using doesn't like the fact that Minio is returning Etag instead of ETag as it is mentioned in the AWS Documentation. To test, we removed Eucalyptus out of the picture and tried to upload an object directly with AWS Java SDK 1.7.1, which also failed in the same way. However, the same upload request works with AWS Java SDK 1.10.x.
+So, to use Minio successfully as an Eucalyptus ObjectStorage backend, we either need a fix in Minio where it returns the correct ETag or when Eucalyptus starts using a newer AWS Java SDK in a future release.
+
+Here are the requests for the fix in Minio: https://github.com/minio/minio/issues/3284
+and for the AWS Java SDK update in Eucalyptus: https://eucalyptus.atlassian.net/browse/EUCA-12955
+
+Update:
+Looks like this was a bug in AWS Java SDK 1.7.1 and fixed in a later release,
+https://github.com/aws/aws-sdk-java/pull/590
```
Comment (NitishT) [inline docs/minio-as-eucalyptus-object-storage.md]: Since the issue is fixed, pls remove these details and mention the AWS SDK version that works.

**Comment by deekoder** [comment]
@NitishT @harshavardhana - Can we close this PR or merge it?

**Comment by koolhead17** [comment]
Wrote to the contributor, he will make the chnages in the doc and we can merge it afterwards.

**Pull request merged**